gapis/resolve/memory.go

```diff
@@ -139,7 +139,10 @@ func filterTypedRanges(ranges []*service.TypedMemoryRange) ([]*service.TypedMemo
 					mostPreciseLevel = preciseLevel
 				}
 			}
-			index = dups[len(dups)-1] + 1
+			index++
+			if skip := dups[len(dups)-1] + 1; skip > index {
+				index = skip
+			}
```

Comment:
Thanks for finding this corner case!! 
After another look, I find I have another thing missed here about array index check. Maybe this part should be further updated with something like:
```suggestion
			if len(dups) > 0 && dups[len(dups)-1] + 1 > index {
				index = dups[len(dups)-1] + 1
			}
```